docs/limitations.md

```diff
@@ -0,0 +1,69 @@
+
+This brings a few limitations and affects your choices of how you can run Rector.
+
+## PHP Language Level
+
+The PHP process that runs Rector code also needs to load and parse your source code. So, you need to run a version of PHP that is both compatible with your project code as well as Rector.
+
+This means you cannot run Rector on PHP 8 to work on a project that uses language features limited to previous PHP versions.
+
+> One example for this is [array and string offset access using curly braces](https://www.php.net/manual/de/migration74.deprecated.php#migration74.deprecated.core.array-string-access-curly-brace).
+
+But also the other way round, you cannot run Rector on PHP 7 to analyze a project with PHP 8 language features, like [constructor property promotion](https://stitcher.io/blog/constructor-promotion-in-php-8).
+
+Together with Rector's minimum PHP 7.3 version requirement, this might limit your options of how to run Rector.
+
+Note that using an older version of Rector might allow you to run a version of PHP <7.3.
+
+## Code Collisions
+
+Since Rector needs to load the code it works on in addition to the code it needs to run itself, conflicts may occur.
+
+One practical example of this is when you're running Rector on a project that uses an older version of Symfony. Since Rector uses a few Symfony Components itself, you may either run into problems when classes are re-declared, or when Rector happens to load and later (by accident) run parts of Symfony from your project installation. In the later case, the mix-up of different versions may lead to generally unpredictable results.
+
+To remedy this situation, [Rector Prefixed](https://github.com/rectorphp/rector-prefixed) has been created. Basically, it takes Rector and all of its dependencies and moves _all_ that code under a unique namespace prefix.
+
+(Add a note here: Drawbacks? Why isn't Rector-prefixed _the_ recommended approach?)
+
+## Installation Options
+
+With these limitations in mind, here are your installation options.
+
+### 1. Direct Composer install
+
+```bash
+composer require rector/rector --dev
+```
+
+Composer will try to find a version of Rector compatible with your current version of PHP. Also, since it will resolve a _single_ version for every dependency your project and Rector have in common, code collisions will not occur.
+
+The downside is that having Rector installed this way might affect the dependency versions your project will be able to use.
+
+### 2. Use Composer to install Rector Prefixed
+
+```bash
+composer require rector/rector-prefixed --dev
+```
+
+Since Rector Prefixed brings along most of the code it needs in a separate namespace, the impact and resulting limitations for your project's requirements are minimal.
+
+As of writing, the only additional dependency requirement is on `phpstan/phpstan`. But apart from that, Composer should be able to install it with whatever packages your project requires.
```

Comment:
for string operation, it may require nette/utils as well.

Reply:
Shouldn’t that be declared as a dependency then?

https://github.com/rectorphp/rector-prefixed/blob/master/composer.json

Reply:
probably need to be added to :

https://github.com/rectorphp/rector/blob/a9ca776ddc7da6dd000c7070681f2c6b64fccac2/scoped/composer.json#L10

@TomasVotruba what do you think?

Reply:
https://github.com/rectorphp/rector-prefixed/tree/master/vendor/nette/utils

`nette/utils` is contained in the prefixed vendors, does that answer it?

Reply:
oh, probably I was trying `rector-prefixed` that doesn't contain `nette/utils`